_posts/2020-02-24-qute.adoc

```diff
@@ -0,0 +1,182 @@
+---
+layout: post
+date:   2020-02-24 00:00 +0100
+author: mkouba
+synopsis: Qute - a templating engine designed specifically to meet the Quarkus needs.
+---
+
+== Qute - (Not Just) Yet Another Templating Engine
+
+[NOTE]
+====
+Qute is an experimental feature. 
+
+There is no guarantee of stability nor long term presence in the platform until the solution matures. 
+
+An https://quarkus.io/guides/qute[introduction guide] and a more comprehensive https://quarkus.io/guides/qute-reference[reference guide] is available.
+====
+
+Let's start with a very good question: _"Why yet another templating engine?"_.
+There are plenty of templating libraries in Java.
+And Quarkus is known to build on top of "Best of Breed Libraries and Standards".
+That's true.
+On the other hand, Quarkus community is also a powerful innovation catalyst.
+And so we decided to start Qute (QUarkus TEmplates) - a templating engine designed specifically to meet the Quarkus needs. 
+We believe that we can bring new ideas even to such an explored area as the templating is.
+
+=== Basic Ideas
+
+Our main goal is to provide an opinionated innovative templating engine.
+But we don't want to reinvent the wheel.
+Instead, we got inspired by existing technologies.
+Just a few examples:
+
+* The https://quarkus.io/guides/qute-reference#syntax-and-building-blocks[syntax] is mainly inspired by https://handlebarsjs.com/[Handlebars] and https://www.dustjs.com/[Dust.js].
+* The https://quarkus.io/guides/qute-reference#include_helper[template inheritance] is inspired by https://en.wikipedia.org/wiki/Facelets[Facelets] and https://docs.djangoproject.com/en/3.0/ref/templates/language/[Django]. 
+* Qute supports the https://en.wikipedia.org/wiki/Elvis_operator[elvis operator] you might be familiar with from https://kotlinlang.org/[Kotlin]. 
```

Comment:
And groovy

Reply:
https://en.wikipedia.org/wiki/Elvis_operator#Languages_supporting_the_Elvis_operator

Should we mention all of them? ;-)

I believe that one reference is enough...

Reply:
i would say if you are going to mention the elvis operator in JVM land refer to groovy over kotlin as groovy popularized it imo.